### PR TITLE
fix(search): fix weird behaving in search bar

### PR DESCRIPTION
**Pull request by tomwwinter**
see issue: #366 

### Visible/Frontend Changes
- [x] show current state of the search is displayed ("Searching", "No Results", ..)
- [x] wait a short time before search input triggers a search (user typing)
- [x] minimum of three characters in search

### Architectural/Backend Changes
- [x] An already running search is cancelled (technically only ignored) and replaced with the new search query (switchMap)

Closing #366

### CODE REVIEW

File: src/app/core/ui/search/search.component.ts [hunk 1]
```diff
@@ -28,9 +37,91 @@ export class SearchComponent implements OnInit {
...
+        switchMap((value) => {
+          return value;
+        })
+      )
+      .subscribe(
+        (result: { queryResults: any; searchTerms: string[] }) => {
+          this.handleSearchQueryResult(result);
+        },
+        (error) => {
+          console.error("SearchQuery failed", error);
```
Comment (sleidig) [inline src/app/core/ui/search/search.component.ts:58]: Let's use "LoggingService" for this to get it properly registered. Possibly also an error to the user through AlertService?

File: src/app/core/ui/search/search.component.ts [hunk 2]
```diff
@@ -28,9 +37,91 @@ export class SearchComponent implements OnInit {
...
+      this.showSearchToolbar = false;
+    }
+  }
+
+  toggleSearchToolbar() {
+    this.showSearchToolbar = !this.showSearchToolbar;
+  }
+
+  searchInputChanged(searchString) {
+    this.searchStringChanged.next(searchString);
```
Comment (sleidig) [inline src/app/core/ui/search/search.component.ts:104]: can we call this directly in the template (and reduce the number of methods in the class here)?

File: src/app/core/ui/search/search.component.html [hunk 3]
```diff
@@ -1,26 +1,89 @@
...
+    #autoResults="matAutocomplete"
+    (optionSelected)="clickOption($event.option)"
+  >
+    <mat-option
+      *ngIf="
+        searchInput.value.length > 0 &&
+        searchInput.value.length < MIN_CHARACTERS_FOR_SEARCH
+      "
+    >
+      <p>Insert at least {{ MIN_CHARACTERS_FOR_SEARCH }} characters</p>
```
Comment (sleidig) [inline src/app/core/ui/search/search.component.html:27]: I would give the "operational" mat-options a different styling than the actual search result items.

maybe something like
```
result-hint {
    background-color: lightgrey;
    font-style: italic;
}
```

File: src/app/core/ui/search/search.component.ts (hunk 1 above)
Comment (tomwwinter) [reply src/app/core/ui/search/search.component.ts]: loggingService and alertService added

File: src/app/core/ui/search/search.component.ts (hunk 2 above)
Comment (tomwwinter) [reply src/app/core/ui/search/search.component.ts]: done.

File: src/app/core/ui/search/search.component.html (hunk 3 above)
Comment (tomwwinter) [reply src/app/core/ui/search/search.component.html:28]: good point, updated the styles - I would stay with the hints

**Pull request merged**